velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -260,9 +278,10 @@ struct FromUnixtimeFunction {
       const core::QueryConfig& config,
       const arg_type<int64_t>* /*unixtime*/,
       const arg_type<Varchar>* format) {
+    legacyTimeParser_ = config.sparkLegacyTimeParser();
     sessionTimeZone_ = getTimeZoneFromConfig(config);
     if (format != nullptr) {
-      setFormatter(*format);
+      setFormatter(*format, legacyTimeParser_);
```

Comment:
Why do we pass legacyTimeParser_ to setFormatter? It seems to have access to it already, no?

Reply:
Thanks. Updated.